### PR TITLE
Added Fastly params in config

### PR DESCRIPTION
**Pull request by vidarl**
This PR makes it possible to set Fastly configuration via environment variables, prepares the semantic configuration and make it a minimal effort for users to configure it once the fastly bundle has been installed

It also renames app/config/env/docker.php → app/config/env/generic.php as the file contains no docker specific stuff.

It might be a discussion if we should put this in platform or platform-ee as fastly will anyway not be available in platform. But recommendation is already partly here anyway ( I added the missing parts too ).

### CODE REVIEW

File: app/config/ezplatform.yml [hunk 1]
```diff
@@ -37,6 +37,6 @@ ezpublish:
               # As we by default enable EzSystemsPlatformHttpCacheBundle which is designed to expire all affected cache
               # on changes, and as error / redirects now have separate ttl, we easier allow ttl to be greatly increased
               default_ttl: '%httpcache_default_ttl%'
-            # HttpCache purge server(s) setting, eg Varnish, for when ezpublish.http_cache.purge_type is set to 'http'.
             http_cache:
+                # HttpCache purge server(s) setting, eg Varnish, for when ezpublish.http_cache.purge_type is set to 'varnish'.
                 purge_servers: ['%purge_server%']
```
Comment (vidarl) [inline app/config/ezplatform.yml]: I am considering also adding the following after this line:

```
                # Enable the following line if you have installed the EzSystemsPlatformFastlyCacheBundle bundle
                # fastly:
                #     service_id: '%fastly_service_id%'
                #     key: '%fastly_key%'
```

File: .env [hunk 2]
```diff
@@ -29,6 +29,12 @@ WEB_HOST=web
 SELENIUM_HOST=selenium
 
 # Enable recommendations by setting valid id, key and uri
+# In order for this to work you also need to have the EzSystemsRecommendationBundle installed
 #RECOMMENDATIONS_CUSTOMER_ID=""
 #RECOMMENDATIONS_LICENSE_KEY=""
 #PUBLIC_SERVER_URI=""
+
+# Enable fastly by setting valid service_id and key
+# In order for this to work you also need to have the EzSystemsPlatformFastlyCacheBundle installed
```
Comment (andrerom) [inline .env]: As it's not apparent by the name, maybe mention it is a EE bundle here. Unlike Reco, it's not public, so no way to look it up and discover you need a license for it on github / packagist.

File: app/config/env/generic.php [hunk 3]
```diff
@@ -90,12 +90,41 @@
...
     $container->setParameter('purge_server', $purgeServer);
 }
 
 if ($value = getenv('HTTPCACHE_DEFAULT_TTL')) {
     $container->setParameter('httpcache_default_ttl', $value);
 }
+
+// EzSystemsRecommendationsBundle settings
+if ($value = getenv('RECOMMENDATIONS_CUSTOMER_ID')) {
+    $container->setParameter('ez_recommendation.default.yoochoose.customer_id', $value);
```
Comment (adamwojs) [inline app/config/env/generic.php]: Nitpick: EzSystemsRecommendationsBundle parameters are out of commit/PR scope. Could you please move those changes to separate commit?

**Comment by bdunogier** [comment]
> It might be a discussion if we should put this in platform or platform-ee as fastly will anyway not be available in platform

We could clarify that by specifying in the file that Fastly support can be obtained with the Enterprise Edition of eZ Platform.

File: app/config/env/generic.php (hunk 3 above)
Comment (bdunogier) [reply app/config/env/generic.php]: An important nitpick :)

File: app/config/ezplatform.yml (hunk 1 above)
Comment (bdunogier) [reply app/config/ezplatform.yml]: Is it really needed ? Uncommenting these lines shouldn't have direct benefits. It uses default settings that come from the built-in parameters.yml.

File: app/config/env/generic.php (hunk 3 above)
Comment (vidarl) [reply app/config/env/generic.php]: Moved to separate commit ( need to rebase in order to get ridd of the fixup!s though)

File: .env (hunk 2 above)
Comment (vidarl) [reply .env]: Fixed

**Comment by vidarl** [comment]
> Unsure about having this in ezsystems/ezplatform as well. Since it requires EE, isn't it easier to have it over there ?
Yeah, maybe it is.. I added here since reco bundle was done like this ( partly at least )...

But if you prefer, I might just rename docker.php -> generic.php here in ezplatform, and do the rest in a platform-ee PR

**Comment by andrerom** [comment]
> But if you prefer, I might just rename docker.php -> generic.php here (+ the RECOMENDATION fix) in ezplatform, and do the rest in a platform-ee PR

maybe that is cleaner indeed, then my comment on clarifying it is for ee only goes somewhat away.

**Comment by vidarl** [comment]
Ok. Closing this PR and creating two new onces...

**Comment by vidarl** [comment]
New PRs:
platform : https://github.com/ezsystems/ezplatform/pull/231
platform-ee : https://github.com/ezsystems/ezplatform-ee/pull/68
